baybe/objectives/desirability.py

```diff
@@ -0,0 +1,135 @@
+"""Functionality for desirability objectives."""
+
+from collections.abc import Sequence
+from functools import partial
+from typing import Callable, Union
+
+import cattrs
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from attrs import define, field
+from attrs.validators import deep_iterable, instance_of, min_len
+from typing_extensions import TypeGuard
+
+from baybe.objectives.base import Objective
+from baybe.objectives.enum import Scalarization
+from baybe.targets.base import Target
+from baybe.targets.numerical import NumericalTarget
+from baybe.utils.basic import to_tuple
+from baybe.utils.numerical import geom_mean
+
+
+def _normalize_weights(weights: Sequence[Union[float, int]]) -> tuple[float, ...]:
```

Comment:
is int strictly necessary? i seem to recall in several places for parameterss some time ago we always had these unions but simply using float works just as well

Reply:
Ah, I wasn't aware of that tbh, [but it seems you are right](https://peps.python.org/pep-0484/#the-numeric-tower)! Will change accordingly 👍🏼

Reply:
I will mark this day in my calendar and print myself a tshirt

Reply:
Please order for the entire team 🚀

Reply:
fixed